### PR TITLE
test/e2e/autoscaling: refactor installNVIDIADriversDaemonSet()

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
 
 /kind cleanup
 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

refactor installNvidiaDriversDaemonSet and rename it to installNVIDIADriversDaemonSet.
/cc @neolit123 
/cc @alejandrox1 

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -999,10 +999,19 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 	})
 })
 
-func installNvidiaDriversDaemonSet(namespace string) {
+func installNVIDIADriversDaemonSet(f *framework.Framework) {
```
Comment (neolit123) [inline test/e2e/autoscaling/cluster_size_autoscaling.go:1002]: i think the old name is fine.
Nvidia is not an abbreviation and the styling Nvidia is also used (next to nVIDIA and NVIDIA):
https://en.wikipedia.org/wiki/Nvidia

we also have structs named like `NvidiaGPUUpgradeTest` already.

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -999,10 +999,19 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+	dsYamlUrl := "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/daemonset.yaml"
+	framework.Logf("Using %v", dsYamlUrl)
+	// Creates the DaemonSet that installs Nvidia Drivers.
+	ds, err := framework.DsFromManifest(dsYamlUrl)
+	framework.ExpectNoError(err)
+	ds.Namespace = f.Namespace.Name
+
+	_, err = f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Create(context.TODO(), ds, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "failed to create nvidia-driver-installer daemonset")
+	framework.Logf("Successfully created daemonset to install Nvidia drivers.")
```
Comment (neolit123) [inline test/e2e/autoscaling/cluster_size_autoscaling.go:1014]: this success line (Logf) can be removed.

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 3]
```diff
@@ -999,10 +999,19 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
 	})
 })
 
-func installNvidiaDriversDaemonSet(namespace string) {
+func installNVIDIADriversDaemonSet(f *framework.Framework) {
 	ginkgo.By("Add daemonset which installs nvidia drivers")
-	// the link differs from one in GKE documentation; discussed with @mindprince this one should be used
-	framework.RunKubectlOrDie(namespace, "apply", "-f", "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/daemonset.yaml")
```
Comment (neolit123) [inline test/e2e/autoscaling/cluster_size_autoscaling.go:1005]: are we removing usage of external calls to `kubectl`?

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 1 above)
Comment (tanjunchen) [reply test/e2e/autoscaling/cluster_size_autoscaling.go:1002]: get  it

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 3 above)
Comment (tanjunchen) [reply test/e2e/autoscaling/cluster_size_autoscaling.go:1005]: currently , we use kubectl in many places , it is difficult to remove kubectl , i  recommend f.ClientSet just here.

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 2 above)
Comment (tanjunchen) [reply test/e2e/autoscaling/cluster_size_autoscaling.go:1014]: sure

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 3 above)
Comment (tanjunchen) [reply test/e2e/autoscaling/cluster_size_autoscaling.go:1005]: I don't know if this is correct.

**Review by alejandrox1 — COMMENTED** [review]
I want to leave this one for @kubernetes/sig-autoscaling-pr-reviews 

This PR touches solely on a e2e test maintained by the SIG and this sort of cleanup is out of scope for testing-commons - I'm not a contributor of sig autoscaling and thus have very little say on this one.

/cc @mwielgus @bskiba

**Comment by bskiba** [comment]
@MaciekPytel can you find someone to take a look?
/assign @MaciekPytel

**Review by fejta — APPROVED** [review]
/lgtm
/approve

**Pull request merged**